subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/logging/TestWorkerProgressListener.java

```diff
@@ -98,7 +101,39 @@ private boolean isDefaultTestClassDescriptor(TestDescriptorInternal testDescript
     private String createProgressLoggerDescription(TestDescriptorInternal testDescriptor) {
         DecoratingTestDescriptor decoratingTestDescriptor = (DecoratingTestDescriptor)testDescriptor;
         DefaultTestClassDescriptor defaultTestClassDescriptor = (DefaultTestClassDescriptor)decoratingTestDescriptor.getDescriptor();
-        return "Executing test " + defaultTestClassDescriptor.getClassName();
+        return "Executing test " + abbreviateJavaPackage(defaultTestClassDescriptor.getClassName());
+    }
+
+    protected String abbreviateJavaPackage(String qualifiedClassName) {
```

Comment:
I'd pull this logic and constants into a separate class. That'll make it much easier to test it. You already made this `protected` for testing assume which is an indication that it should be pulled out.